### PR TITLE
Add CullDistance BuiltIn

### PR DESCRIPTION
**Pull request by Gordon-F**

### CODE REVIEW

File: src/valid/interface.rs [hunk 1]
```diff
@@ -125,6 +125,19 @@ impl VaryingContext<'_> {
                             _ => false,
                         },
                     ),
+                    Bi::CullDistance => (
```
Comment (kvark) [inline src/valid/interface.rs:128]: can we merge this path with `Bi::ClipDistance`?

File: src/back/msl/writer.rs [hunk 2]
```diff
@@ -2074,6 +2080,13 @@ impl<W: Write> Writer<W> {
                             first_time: true,
                         };
                         let binding = binding.ok_or(Error::Validation)?;
+                        // Cull Distance is not supported in Metal.
+                        // But we can't return UnsupportedBuiltIn error to user.
+                        // Because otherwise we can't generate msl shader from any glslang SPIR-V shaders.
+                        // glslang generates gl_PerVertex struct with gl_CullDistance builtin inside by default.
+                        if *binding == crate::Binding::BuiltIn(crate::BuiltIn::CullDistance) {
```
Comment (kvark) [inline src/back/msl/writer.rs:2087]: let's issue `log::warn!("Ignoring CullDistance")`

**Review by kvark — COMMENTED** [review]
Looks great!
Just a few nits

**Comment by Gordon-F** [comment]
> Just a few nits

Should be ready to merge now.

File: src/back/msl/mod.rs [hunk 3]
```diff
@@ -135,6 +135,8 @@ pub enum Error {
     FeatureNotImplemented(String),
     #[error("module is not valid")]
     Validation,
+    #[error("BuiltIn {0:?} is not supported")]
+    UnsupportedBuiltIn(crate::BuiltIn),
```
Comment (kvark) [inline src/back/msl/mod.rs:139]: why are we having this error if the MSL backend just works around this builtin anyway?

**Review by kvark — APPROVED** [review]

**Pull request merged**